.github/workflows/build.yml

```
@@ -0,0 +1,121 @@
+name: Build
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+jobs:
+  prep:
+    name: Prepare the build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Update the 'latest' tag
+        if: github.ref == 'refs/heads/main'
+        run: |
+           git config user.name github-actions
+           git config user.email github-actions@github.com
+           git push origin :refs/tags/latest || true
+           git tag -m latest -fa latest || true
+           git push origin main --tags
+  build:
+    name: Build on ${{ matrix.os }} with Java ${{ matrix.java }}
+    needs: prep
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
```

This might be one were the separate action is slightly better, because you probably want to see all the failures during a PR's CI, but fast-fail when pushing the built tags. Could probably get something similar with a condition, or just leave it as-is
```suggestion
      fail-fast: false
```